### ISSUE TITLE
Link to item parsed incorrectly if contains ] symbol in name

### ISSUE THREAD

**Issue by Greatshock** [foundryvtt/dnd5e#4873]
In Russian-speaking community it is common to write names for actors and items in the following format:
`nameInRussian [nameInEnglish]`

The reason for that is that we do not have official terms in Russian, so everyone translates them as they want and write original name in square brackets.
However, if you try to embed an item with such name in journal or another item's description, it will be parsed incorrectly:
`[[/item Тест [Test]]]`

<img width="117" alt="Screenshot 2024-12-12 at 11 41 20 PM" src="https://github.com/user-attachments/assets/06d42ec3-4e92-40a2-b846-334974a38884" />

I also tried to put space before the closing `]]`, but it is not recognized as a link in this case. 
<img width="149" alt="Screenshot 2024-12-12 at 11 41 47 PM" src="https://github.com/user-attachments/assets/abb1157e-5a2a-4997-8118-f2b07fb2a3b2" />

It would also be great if such syntax would work:
`[[/item Атака [Attack] activity="Some Special Attack"]]{My Attack}`

Please consider rewriting parser to match closing symbols from the end.

**Comment by Greatshock** [comment]
@arbron hi, could you please check if this can be fixed in 4.2.0 or won't be fixed at all?

**Comment by roth-michael** [comment]
Would have to do some more testing with this but I _think_ this may be solvable by changing the `(?<config> [^\\]]+)?]]` part of the regex to be `(?<config> .+)?]](?!\\])`,
i.e.
```diff
- pattern: new RegExp(`\\[\\[/(?<type>${stringNames.join("|")})(?<config> [^\\]]+)?]](?:{(?<label>[^}]+)})?`, "gi"),
+ pattern: new RegExp(`\\[\\[/(?<type>${stringNames.join("|")})(?<config> .+)?]](?!\\])(?:{(?<label>[^}]+)})?`, "gi"),
```
